### ISSUE TITLE
Event stream producing no data

### ISSUE THREAD

**Issue by JoelOtter** [Netflix/Hystrix#758]
I've got Hystrix set up with the event stream on a project. The stream is running, but I don't get any information back when `curl`ing it other than:

```
ping: 

ping:

ping:

```
etc. I'm not getting much in the way of errors.

I'm using hystrix-core and hystrix-metrics-event-stream versions 1.4.4 each, so I can't imagine this is an incompatibility issue. I'm also able to pull metrics out of the HystrixMetrics using, for instance:

`HystrixCommandMetrics.getInstance(this.commandKey).getHealthCounts().getTotalRequests());`

I'm not sure what's causing this.

**Comment by mattrjacobs** [comment]
I don't have an answer off the top of my head.  I'll try to replicate what you're seeing and report back.

**Comment by JoelOtter** [comment]
Turns out I actually had the event stream and the HystrixCommands running in separate Tomcat artifacts. Working perfectly after moving my web.xml around. Thanks, and apologies for any inconvenience caused!

**Comment by mattrjacobs** [comment]
No problem - glad you got it sorted out.

**Comment by vijayparashar12** [comment]
I Have configured my web app as described in README of matrix-event-stream 

Here is my web.xml snippet

<servlet>
        <description></description>
        <display-name>HystrixMetricsStreamServlet</display-name>
        <servlet-name>HystrixMetricsStreamServlet</servlet-name>
        <servlet-class>com.netflix.hystrix.contrib.metrics.eventstream.HystrixMetricsStreamServlet</servlet-class>
         <load-on-startup>1</load-on-startup>
    </servlet>

    <servlet-mapping>
        <servlet-name>HystrixMetricsStreamServlet</servlet-name>
        <url-pattern>/hystrix.stream</url-pattern>
    </servlet-mapping>

My Application Logs show 

16:22:26.206 [http-nio-8008-exec-1] DEBUG c.n.h.c.m.e.HystrixMetricsPoller - Starting HystrixMetricsPoller
16:22:26.211 [http-nio-8008-exec-1] DEBUG c.n.h.c.m.e.HystrixMetricsStreamServlet - Starting poller
16:22:56.563 [http-nio-8008-exec-2] DEBUG c.n.h.c.m.e.HystrixMetricsPoller - Starting HystrixMetricsPoller
16:22:56.564 [http-nio-8008-exec-2] DEBUG c.n.h.c.m.e.HystrixMetricsStreamServlet - Starting poller


**But when I curl for event stream url it gives nothing.**

curl -v http://192.168.45.212:8008/activities-api/hystrix.stream
Hostname was NOT found in DNS cache
Trying 192.168.45.212...
Connected to 192.168.45.212 (192.168.45.212) port 8008 (#0)
 GET /activities-api/hystrix.stream HTTP/1.1
 User-Agent: curl/7.38.0
 Host: 192.168.45.212:8008
 Accept: */*
 
 HTTP/1.1 200 OK
 Server Apache-Coyote/1.1 is not blacklisted
 Server: Apache-Coyote/1.1
 Cache-Control: no-cache, no-store, max-age=0, must-revalidate
 Pragma: no-cache
 Content-Type: text/event-stream;charset=UTF-8
 Transfer-Encoding: chunked
 Date: Fri, 03 Jun 2016 10:52:26 GMT
 >

**Comment by mattrjacobs** [comment]
The HTTP 200 is a good sign.  On the application that the stream is coming from, have there been any commands executed?  If not, then there's no data to show yet.

**Comment by vijayparashar12** [comment]
@mattrjacobs yes , I executed a command around 5000 times in 60 seconds. also, I have added this code in my controller just after the command is executed.(Just to be double sure)

```
HystrixCommandMetrics instance = HystrixCommandMetrics.getInstance(HystrixCommandKey.Factory.asKey(HEALTH_CHECK));// , HystrixCommandGroupKey.Factory.asKey("Local"));
        log.info("total =>" + instance.getHealthCounts().getTotalRequests());
        log.info("error =>" + instance.getHealthCounts().getErrorCount());
```
But in Logs ..
```
2016-06-03 23:09:43,760 [http-nio-8008-exec-1] INFO [ctLocal.controller.HealthCheckController]  - total =>0
2016-06-03 23:09:43,784 [http-nio-8008-exec-1] INFO [ctLocal.controller.HealthCheckController]  - error =>0

```

Here is how my command class looks like

```
public class Command extends HystrixCommand<Map<String, String>> {

    Command() {
        super(Setter
                .withGroupKey(HystrixCommandGroupKey.Factory.asKey("Local"))
                .andCommandKey(HystrixCommandKey.Factory.asKey(HealthCheckController.HEALTH_CHECK))
                .andCommandPropertiesDefaults(
                        HystrixCommandProperties.Setter().withCircuitBreakerEnabled(true).withCircuitBreakerErrorThresholdPercentage(50).withExecutionTimeoutInMilliseconds(5000)
                                .withFallbackEnabled(true)
                // .withMetricsRollingStatisticalWindowInMilliseconds(60 * 1000)
                ));
    }

    @Override
    protected Map<String, String> run() throws Exception {
        RestResponse restResponse = RestUtil.get("http://172.16.53.133:9200/_cluster/health", null);
        return JsonUtil.getObjectMapper().readValue(restResponse.getMessage(), new TypeReference<Map<String, String>>() {
        });
    }

}
```

**Is There any configuration that I am missing?**

**Comment by mattrjacobs** [comment]
Hmmm, hard to say what's happening.  Some things that would help diagnose where the data is going missing:

1) What commands are configured on your JVM?
`new HystrixConfigurationStream(100).observe().toBlocking().single()`

2) What commands does the metrics subsystem of Hystrix think you've executed?
`HystrixCommandMetrics.getInstances().map(m -> m.getCommandKey())`

The first should look into the data structure that Hystrix sets up each time it encounters a new command and is the canonical source of all commands on the JVM.  The second gets set up as that command starts executing and producing metrics.  Seeing the result of those will help understand what's happening.

**Comment by ianaz** [comment]
Hi, same issue with the stream only printing out "ping".

> HystrixCommandMetrics.getInstances() 

And getting the first healthCounts()

> HystrixCommandMetrics.getInstances().iterator().next().getHealthCounts()

 "totalCount" increases for each call so until there is fine

Any suggestion where to check next?
Using hystrix-core and hystrix-stream both on 1.5.9, Tomcat 7

**Comment by ianaz** [comment]
Solved by downgrading from 1.5.9 to **1.4.3**

**Comment by mattrjacobs** [comment]
@ianaz I'm not 100% sure what your case shares with the descriptions above, and what's different.  If you'd like help investigating your case, please open a new issue.  FWIW, we're using the 1.5.9 jars in prod, so they should work
